plugins/janus_audiobridge.c

```diff
@@ -3230,9 +3291,39 @@ static json_t *janus_audiobridge_process_synchronous_request(janus_audiobridge_s
 		json_object_set_new(response, "permanent", save ? json_true() : json_false());
 		JANUS_LOG(LOG_VERB, "Audiobridge room destroyed\n");
 		goto prepare_response;
+	} else if(!strcasecmp(request_text, "enable_recording")) {
+		JANUS_VALIDATE_JSON_OBJECT(root, record_parameters,
+			error_code, error_cause, TRUE,
+			JANUS_AUDIOBRIDGE_ERROR_MISSING_ELEMENT, JANUS_AUDIOBRIDGE_ERROR_INVALID_ELEMENT);
+		if(error_code != 0)
+			goto prepare_response;
+		json_t *record = json_object_get(root, "record");
+		gboolean recording_active = json_is_true(record);
+		JANUS_LOG(LOG_VERB, "Enable Recording : %d \n", (recording_active ? 1 : 0));
+		/* Lookup room */
+		janus_mutex_lock(&rooms_mutex);
+		janus_audiobridge_room *audiobridge = NULL;
+		error_code = janus_audiobridge_access_room(root, TRUE, &audiobridge, error_cause, sizeof(error_cause));
+		if(error_code != 0) {
+			JANUS_LOG(LOG_ERR, "Failed to access audiobridge\n");
+			janus_mutex_unlock(&rooms_mutex);
+			goto prepare_response;
+		}
+		janus_mutex_unlock(&rooms_mutex);
+		/* Set recording status */
+		gint room_prev_recording_active = recording_active ? 1 : 0;
+		if (room_prev_recording_active != g_atomic_int_get(&audiobridge->record)) {
+			/* Room recording state has changed */
+			JANUS_LOG(LOG_VERB, "Recording status changed : prev = %d curr = %d \n", g_atomic_int_get(&audiobridge->record), recording_active);
```

Comment:
Debug style: there should be no space before the semicolon (`changed: `), or empty spaces at the end of a sentence. The spaces around the `=` are unneeded as well, and values should be comma separated (`prev=%d, curr=%d\n`).